File: Mar-18/18-03-18/c++/01-Sumits-love-for-maths.cpp
#include<bits/stdc++.h>
#include <boost/math/common_factor.hpp>

using namespace std;

int main()
{
    int t;
    cin>>t;
    while(t--)
    {
        long n,a,b,c;
        cin>>n>>a>>b>>c;
        
        long total=(n/a)+(n/b)+(n/c);
        total -= (n/boost::math::lcm(a,b)); 
        total -= (n/boost::math::lcm(a,c)); 
        total -= (n/boost::math::lcm(b,c)); 
        total -= (n/boost::math::lcm(a,boost::math::lcm(b,c)));
        cout<<total<<endl;
    }
}

File: Mar-18/18-03-18/python/01-Sumits-love-for-maths.py
def GCD(a,b):
    if(b==0):
        return a
    else:
        return GCD(b,a%b)
def LCM(a,b):
    return (a*b)//GCD(a,b)
    
for _ in range( int(input()) ):
    
    n,a,b,c = map(int,input().split())
    total = (n//a) + (n//b) + (n//c)  
    total -= (n//LCM(a,b))
    total -= (n//LCM(a,c))
    total -= (n//LCM(b,c))
    total += (n//LCM(a,LCM(b,c)))
    print(int(total))
    
    
